### PR TITLE
Update to Arrow-6.0.0

### LINKED ISSUE #222: Arrow 6.0 has been released
* Parquet version enums have changed.
* Null statistics might have been fixed in more cases (check unit tests)
* Decimal statistics might have been fixed in more cases (#68 and unit tests)

### PR DESCRIPTION
**Pull request by jgiannuzzi**
Closes #222.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,7 @@ if (MSVC)
 
 	add_compile_options("/MP") # Compile files in parallel.
 	add_compile_options("/WX") # Threat warnings as errors.
+	add_compile_options("/Zc:__cplusplus") # Enable standard __cplusplus macro.
```
Comment (GPSnoopy) [inline CMakeLists.txt:36]: Why is this part needed ?

Comment (jgiannuzzi) [reply CMakeLists.txt:36]: Without this, the build on Windows fails because the `__cplusplus` macro is not defined by default by MSVC: https://github.com/G-Research/ParquetSharp/runs/4137772529?check_suite_focus=true#step:18:100

Comment (GPSnoopy) [reply CMakeLists.txt:36]: Highly suspicious. This macro should always be defined for C++ compilation:
https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170&viewFallbackFrom=vs-2017

I think the problem comes from arrow in arrow/cpp/src/arrow/util/launder.h:
`#if __cplusplus >= 201703L`

I don't think that the actual value of __cplusplus can be relied on in MSVC. For example, even though I'm compiling for C++20, I get `__cplusplus = 199711` in Visual Studio 2022.

Looks like you came to the same conclusion:
/Zc:__cplusplus : Enable the __cplusplus macro to report the supported standard (off by default).

I suggest you raise this with Ian.

Comment (jgiannuzzi) [reply CMakeLists.txt:36]: Yes, that is the exact reason I had to define `/Zc:__cplusplus`. It's quite unfortunate, but Visual Studio will define `__cplusplus` wrongly by default (See https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/).

I did mention it to Ian [here](https://github.com/microsoft/vcpkg/pull/21113#issuecomment-967006107), though I'm not sure this is a problem with the vcpkg port of Arrow, but rather a quirk of MSVC that we were lucky enough to not encounter until Arrow 6.0.0.

Comment (GPSnoopy) [reply CMakeLists.txt:36]: Worth adding this URL as a comment above this line in the CMake file then

Comment (jgiannuzzi) [reply CMakeLists.txt:37]: Great point, thanks!

**Comment by jgiannuzzi** [comment]
Closed in favour of #247.
